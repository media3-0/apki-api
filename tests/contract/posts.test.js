const Lokka = require('lokka').Lokka;
const Transport = require('lokka-transport-http').Transport;

const client = new Lokka({
  transport: new Transport('http://localhost:9778/graphql'),
});

describe('Query', () => {
  describe('allPosts', () => {
    it('no variables', async () => {
      const result = await client.query(`
        {
          allPosts {
            id
            author
            title
            content
          }
        }
      `);
      expect(result).toMatchSnapshot();
    });
    describe('with variables', () => {
      it('limit', async () => {
        const result = await client.query(`
          {
            allPosts(limit: 2) {
              id
              author
              title
              content
            }
          }
        `);
        expect(result).toMatchSnapshot();
      });
    });
  });
});
