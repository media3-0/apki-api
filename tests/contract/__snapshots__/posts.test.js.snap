exports[`Queries allPosts with limit 1`] = `
Object {
  "allPosts": Array [
    Object {
      "author": "Mietek",
      "content": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus fringilla quis ipsum ut hendrerit. Etiam rutrum elementum urna, et fringilla lacus dapibus sed.",
      "title": "Lorem ipsum",
    },
    Object {
      "author": "Andrzej",
      "content": "Aliquam eu mollis felis. Maecenas quis pulvinar neque. Donec arcu ex, ultrices sit amet nunc ac, interdum dapibus nunc.",
      "title": "Maecenas mattis",
    },
  ],
}
`;

exports[`Queries allPosts without params 1`] = `
Object {
  "allPosts": Array [
    Object {
      "author": "Mietek",
      "content": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus fringilla quis ipsum ut hendrerit. Etiam rutrum elementum urna, et fringilla lacus dapibus sed.",
      "title": "Lorem ipsum",
    },
    Object {
      "author": "Andrzej",
      "content": "Aliquam eu mollis felis. Maecenas quis pulvinar neque. Donec arcu ex, ultrices sit amet nunc ac, interdum dapibus nunc.",
      "title": "Maecenas mattis",
    },
    Object {
      "author": "Zdzisiek",
      "content": "Suspendisse eget tempor sem, eu lobortis tortor. Donec quis sem sapien. Pellentesque nec massa elit. Curabitur sagittis nibh diam, ut volutpat tortor rutrum vel.",
      "title": "Praesent laoreet dui at dui",
    },
    Object {
      "author": "Andrzej",
      "content": "In vitae congue quam. Aenean vel porttitor neque. Etiam non neque vel elit luctus euismod eget non dui. In venenatis arcu dictum elit tristique dictum. Fusce vel mauris vel lorem facilisis egestas. Sed nec interdum orci.",
      "title": "Proin suscipit non quam",
    },
  ],
}
`;
